src/MongoId.php

```diff
@@ -178,13 +181,15 @@ public function getInc()
      *   string consisting of exactly 24 hexadecimal characters; otherwise,
      *   FALSE is returned.
      */
-    public static function isValid($id)
+    public static function isValid($value)
     {
-        if (!is_string($id)) {
+        if (get_class($value) == self::class) {
```

Comment:
Maybe, I have to admit that I don't remember if I chose get_class() for a specific reason or not, your solution probably just works fine.

Also being honest, I moved to official mongodb driver/library, even if they are not completely stable they are made by the mongo team.